### PR TITLE
Refactor horizontalbarchart web component

### PR DESCRIPTION
**Pull request by krkshitij**
- Move legends and tooltip to template

### CODE REVIEW

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.template.ts [hunk 1]
```diff
@@ -14,9 +15,45 @@ getGroupper(tabsterCore);
...
+              </button>`,
+            )}
+          </div>
+        `,
+      )}
+      ${when(
+        x => !x.hideTooltip && x.tooltipProps.isVisible,
+        html<T>`
+          <div class="tooltip" style="left: ${x => x.tooltipProps.xPos}px; top: ${x => x.tooltipProps.yPos}px">
+            <div class="tooltipline" style="border-left:4px solid ${x => x.tooltipProps.color};">
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.template.ts:50]: the style can move to the class if it is static

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.ts [hunk 2]
```diff
@@ -45,14 +44,38 @@ export class HorizontalBarChart extends FASTElement {
...
 
-  @attr
+  @observable
   public uniqueLegends: ChartDataPoint[] = [];
 
   @attr
   public hideRatio = false;
 
+  @attr({ attribute: 'hide-legends', mode: 'boolean' })
+  public hideLegends?: boolean;
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.ts:54]: add a default value for the attribute

**Review by AtishayMsft — APPROVED** [review]

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.ts (hunk 2 above)
Comment (krkshitij) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.ts:54]: added in https://github.com/microsoft/fluentui/pull/33199

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.template.ts (hunk 1 above)
Comment (krkshitij) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.template.ts:50]: here, the color changes dynamically based on the element being hovered over or focused on

**Pull request merged**